services/server/server.go

```diff
@@ -596,6 +599,16 @@ func (s *Server) ChangeSettings(ctx context.Context, req *serverpb.ChangeSetting
 		return nil, err
 	}
 
+	if req.EnableStt {
```

Comment:
I think it's better to move it below `s.UpdateConfiguration` and `isAgentsStateUpdateNeeded`. Then we will be sure that we start checks only when an update is successful.

Reply:
done, waiting for the FB to complete